datajoint/diagram.py

```diff
@@ -300,6 +300,36 @@ def _make_graph(self):
             nx.relabel_nodes(graph, mapping, copy=False)
             return graph
 
+        @staticmethod
+        def _stringify_and_encapsulate_edge_attributes(graph):
+            """
+            Modifies the `nx.Graph`'s edge attribute `attr_map` to be a string representation
+            of the attribute map, and encapsulates the string in double quotes.
+            Changes the graph in place.
+
+            Implements workaround described in
+            https://github.com/pydot/pydot/issues/258#issuecomment-795798099
+            """
+            for u, v, *_, edgedata in graph.edges(data=True):
+                if "attr_map" in edgedata:
+                    graph.edges[u, v]["attr_map"] = '"{0}"'.format(edgedata["attr_map"])
```

Comment:
@tabedzki Re: [this comment](https://github.com/datajoint/datajoint-python/pull/1176#issuecomment-2334732787)

>  Those error messages for test_repr_svg and test_make_image indicate to me that they are not using the patched version of networkx that is currently only available on github and not through pypa.

I understand that the dev release deprecates the colon check that is no longer necessary in modern pydot. We're able to easily support earlier versions as well by also encapsulating the  `attr_map: Dict[str, str]` edge attribute in `"`.